extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js

```diff
@@ -1416,15 +1443,9 @@ export class AmpLightboxGallery extends AMP.BaseElement {
       element.appendChild(timestampDiv);
     }
 
-    const closeGalleryAndShowTargetSlide = event => {
-      this.closeGallery_();
-      this.currentElemId_ = thumbnailObj.element.lightboxItemId;
-      dev().assert(this.carousel_).getImpl()
-          .then(carousel => carousel.showSlideWhenReady(this.currentElemId_));
-      this.updateDescriptionBox_();
-      event.stopPropagation();
-    };
-    element.addEventListener('click', closeGalleryAndShowTargetSlide);
+    const handleThumbnailClick = this.generateThumbnailClickHandler_(
+        thumbnailObj.element.lightboxItemId);
+    element.addEventListener('click', handleThumbnailClick.bind(this));
```

Comment:
It looks like you're creating a new function to save the `id` parameter, but that is a lot of indirection. Instead of returning a function from `generateThumbnailClickHandler`, use the class method to do work, then call it from the fat-arrow function and pass the id.

```js
handleThumbnailClick_(event, id) {
  event.stopPropagation();
  Promise.all([
    this.closeGallery_(),
    dev().assert(this.carousel_).getImpl(),
  ]).then(values => {
    this.currentElemId_ = id;
    const carouselImpl = values[1];
    carouselImpl.showSlideWhenReady(this.currentElemId_);
        this.updateDescriptionBox_();
  });
}
 
/* ... */

element.addEventListener('click', 
    event => this.handleThumbnailClick_(event, thumbnailObj.element.lightboxItemId));
```